sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -290,4 +297,15 @@ protected String visitTrim(String direction, String[] inputs) {
       return "TRIM(" + direction + " " + inputs[1] + " FROM " + inputs[0] + ")";
     }
   }
+
+  protected String visitExtract(String field, String source) {
+    switch (field) {
+      case "DAY_OF_WEEK":
+        return "(EXTRACT(ISO_DAY_OF_WEEK FROM " + source + ") % 7)+ 1";
```

Comment:
the field names are already user-facing (DS v2 API). We should do the translation earlier, when converting catalyst expressions to v2 expressions.

Reply:
n different databases, the ISO_DAY_OF_WEEK is different.
In postgresql uses `isodow `.
In snowflake uses `dow_iso ` .
In H2 uses `ISO_DAY_OF_WEEK `.

Reply:
Then I think `DAY_OF_WEEK` is fine. Different dialects must override it anyway.